stl/inc/source_location

```diff
@@ -0,0 +1,64 @@
+// source_location standard header (core)
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _SOURCE_LOCATION_
+#define _SOURCE_LOCATION_
+#include <yvals_core.h>
+#if _STL_COMPILER_PREPROCESSOR
+#if !_HAS_CXX20 && !defined(__clang__)

```

Comment:
```suggestion
#if !_HAS_CXX20 || !defined(__clang__)
```

Reply:
Haha! Early morning me thinking I'm fixing something https://github.com/microsoft/STL/pull/664/commits/d3f4c9ec81591e73a57a8e613198acfdaac1066a

Reply:
From some quick-and-dirty checks, it looks like MSVC supports the builtins now, and is only lacking `consteval`. So ideally this would be `!_HAS_CXX20 || !defined(__cpp_consteval)`, except that Clang doesn't define `__cpp_consteval` despite implementing `consteval` (the feature-test macro was added to the C++20 working draft fairly late). So let's go with:
```suggestion
#if !_HAS_CXX20 || !(defined(__cpp_consteval) || defined(__clang__)) // TRANSITION, LLVM-45745
```
where LLVM-45745 is a bug nicely asking the clang devs to define the macro.